server/jetty-11/src/main/java/io/deephaven/server/jetty11/JettyConfig.java

```diff
@@ -134,9 +145,31 @@ public static Builder buildFromConfig(Configuration config) {
         if (maxHeaderRequestSize != null) {
             builder.maxHeaderRequestSize(Integer.parseInt(maxHeaderRequestSize));
         }
+        builder.allowedHttpMethods(allowedHttpMethods);
+        builder.extraHeaders(extraHeaders);
         return builder;
     }
 
+    private static Map<String, String> readExtraHeaders(Configuration config) {
+        Map<String, String> extraHeaders = new HashMap<>();
+        final List<String> headerKeys = new ArrayList<>();
+
+        config.getProperties(EXTRA_HEADERS).forEach((suffixArg, valueArg) -> {
+            final String suffix = (String) suffixArg;
+            if (suffix != null && suffix.endsWith(".enabled") && valueArg != null
+                    && Boolean.parseBoolean(valueArg.toString())) {
+                headerKeys.add(suffix.substring(0, suffix.length() - ".enabled".length()));
+            }
+        });
+        for (String headerKey : headerKeys) {
+            final String headerValue = config.getStringWithDefault(EXTRA_HEADERS + headerKey + ".value", null);
+            if (headerValue != null) {
```

Comment:
Is a null value an indication of an error that should be raised?  I think if you say something is enabled, but then do not define it the most likely situation is a typo that you then need to hunt down.

Reply:
I can add that, but this is poached directly from DHE's envoy handling, so should be updated there as well.